wtransport-proto/src/error.rs

```diff
@@ -53,6 +53,16 @@ pub enum ErrorCode {
 }
 
 impl ErrorCode {
+    /// Converts this stream [`ErrorCode`] to HTTP/3 error code.
+    /// <https://ietf-wg-webtrans.github.io/draft-ietf-webtrans-http3/draft-ietf-webtrans-http3.html#section-4.3)>
+    pub fn to_http3(self) -> VarInt {
+        let code = self.to_code();
```

Comment:
Upon reading all relevant, it appears that the conversion is only for WT **application** error codes and not any of the existing `ErrorCode` variants, all of which are in the space of valid HTTP/3 error codes.

This can be seen as follows:
- Most of the variants are in the `h3_error_codes` module and numerically match HTTP/3 error code entries in https://datatracker.ietf.org/doc/html/draft-ietf-quic-http#iana-error-table
- For the variant in the `qpack_error_codes` namespace, it numerically matches an HTTP/3 error code entry in https://www.rfc-editor.org/rfc/rfc9204.html
- For the variants in the `wt_error_codes` module, they numerically match HTTP/3 error code entries in https://ietf-wg-webtrans.github.io/draft-ietf-webtrans-http3/draft-ietf-webtrans-http3.html#name-http-3-error-code-registrat

In other words, if we wanted application error codes, they would work like this:
```rust
enum ErrorCode{
    Datagram,
    NoError,
    ...
    Application(u8),
}

impl ErrorCode {
    pub fn to_code(self) -> VarInt {
        match self {
            ErrorCode::Datagram => h3_error_codes::H3_DATAGRAM_ERROR,
            ErrorCode::NoError => h3_error_codes::H3_NO_ERROR,
            ...
            ErrorCode::Application(n) => to_http3(n),
        }
    }
}
```

But I'm not sure `Applicatinon` would be a useful variant. Instead, I suggest looking into whether `reset(error_code)` should be a `u8` and go through the `to_http3` conversion:
https://github.com/BiagioFesta/wtransport/blob/4fcb949c2c97d9b13b784a1761001fe2b2213e7a/wtransport/src/stream.rs#L83-L93

Reply:
There is one thing I'm unsure about:
- https://ietf-wg-webtrans.github.io/draft-ietf-webtrans-http3/draft-ietf-webtrans-http3.html#name-resetting-data-streams implies there are 2^32 web transport application errors
- https://www.ietf.org/archive/id/draft-ietf-webtrans-http3-02.html#name-http-3-error-code-registrat implies there are only 2^8 web transport application errors

I assume this is because these are two different versions of the spec.